File: debate-engine/src/agents/moderator.ts
import { z } from "zod";
import type { AgentInput, ModeratorResponse } from "../types/index.js";
import { BaseAgent } from "./base.js";
import { ModeratorResponseSchema } from "../llm/responseParser.js";
import { buildModeratorSystemPrompt, buildModeratorUserMessage } from "../prompts/moderator.prompt.js";

export class ModeratorAgent extends BaseAgent<ModeratorResponse> {
  protected readonly role = "moderator";
  protected readonly schema: z.ZodType<ModeratorResponse> = ModeratorResponseSchema;

  protected buildSystemPrompt(input: AgentInput): string {
    return buildModeratorSystemPrompt(input);
  }

  protected buildUserMessage(input: AgentInput): string {
    return buildModeratorUserMessage(input);
  }
}

File: debate-engine-py/src/agents/moderator.py
from src.agents.base import BaseAgent
from src.llm.router import ModelRouter
from src.prompts.moderator import build_moderator_system_prompt, build_moderator_user_message
from src.schemas.agents import ModeratorResponse
from src.schemas.debate import DebateState


class ModeratorAgent(BaseAgent):
    role = "moderator"

    def __init__(self, router: ModelRouter):
        super().__init__(router)

    async def run(self, state: DebateState) -> tuple[ModeratorResponse, dict]:
        system_prompt = build_moderator_system_prompt(state)
        user_message = build_moderator_user_message(state)
        response, usage = await self.invoke(system_prompt, user_message, ModeratorResponse)
        return response, usage
